### ISSUE TITLE
blitbuffer paintRect and fill alpha values

### ISSUE THREAD

**Issue by mapinguari** [koreader/koreader-base#1445]
I ideally would like to be able to paint a rect/fill a buffer with a colour value that contains an alpha. But it looks like both the [C](https://github.com/koreader/koreader-base/blob/ab2fa6f3aedb3e9a799a10df39b3bd5346de3f14/blitbuffer.c#L329-L350) and [lua](https://github.com/koreader/koreader-base/blob/ab2fa6f3aedb3e9a799a10df39b3bd5346de3f14/ffi/blitbuffer.lua#L1533-L1539) implementations of doing this ignore the alpha or just don't accept a parameter for it. Is this expected behaviour?

**Comment by NiLuJe** [comment]
Our preferred output format is Y8, so alpha is and always has been a second thought ;).

(IIRC, I specifically tailored paintRect to take an Y8 color as input).

**Comment by NiLuJe** [comment]
Assuming the' C blitter is disabled, the Lua blitter accepts a polymorphic `Color` as input, so it *should* handle alpha.

It would be fairly gimmicky, though, since the C blitter is only disabled on truly terrible platforms (or maybe none at all, these days).

**Comment by mapinguari** [comment]
What is Y8? (sorry)

Sadly the lua blitter does this:
`            -- Handle invert...
            local v = value:getColor8()
            if self:getInverse() == 1 then v = v:invert() end
`

so ultimately ignores the alpha values (and any rgb while it is at it).

I guess I can just can set `use_cblitbuffer` to `false` and then set it back after.

**Comment by NiLuJe** [comment]
> What is Y8? (sorry)
> 
> Sadly the lua blitter does [this](https://github.com/koreader/koreader-base/blob/ab2fa6f3aedb3e9a799a10df39b3bd5346de3f14/ffi/blitbuffer.lua#L1510-L1512) so ultimately ignores the alpha values (and any rgb while it is at it).
> 

Right, that's mostly to avoid inconsistencies with the C blitter, as that caused hard-to-debug issues in the past ;).

It could technically only do that in codepaths where it matters, while just leaving the invert on top.

TL;DR: You're probably better off writing a new, custom method altogether, because a change to paintRect is unlikely to get merged ;).

**Comment by mapinguari** [comment]
Ok, I will do that then. Thank you
